### PR TITLE
fix #282275: Crash when changing Time Signature in parts

### PR DESCRIPTION
**Pull request by songchaow**
See https://musescore.org/en/node/282275.

### CODE REVIEW

**Comment by jthistle** [comment]
Looks good, but watch your indentation. The last brace should be an extra six spaces indented.

File: libmscore/edit.cpp [hunk 1]
```diff
@@ -418,6 +418,13 @@ bool Score::rewriteMeasures(Measure* fm, Measure* lm, const Fraction& ns, int st
                   }
             return true;
             }
+      // transform fm and lm to measures in masterScore,
+      // if they are originlly in parts
```
Comment (RobFog) [inline libmscore/edit.cpp]: * originally

**Comment by mattmcclinch** [comment]
The fact that `fm->score()->isMaster()` turns out to be false points to an error elsewhere in the code. [This](https://github.com/mattmcclinch/MuseScore/commit/ab4704cf444a389381d711e384292b7d149bd331) corrects that error, and is all that is needed to fix this bug.

**Comment by songchaow** [comment]
> The fact that `fm->score()->isMaster()` turns out to be false points to an error elsewhere in the code. [This](https://github.com/mattmcclinch/MuseScore/commit/ab4704cf444a389381d711e384292b7d149bd331) corrects that error, and is all that is needed to fix this bug.

You are right, though I had thought about this.

The reason why I haven't passed `mf`(from full score) so early, was that, I thought the `fm`(from part score) might be used when we are adding local time signatures to part score.
In this scenario, the `staffIdx` passed to `rewriteMeasures(fm, ns, local ? staffIdx : -1)` will be valid, and indicate the index of the staff in **part score**, not **full score**. So if we're passing `mf`(from full score), the `staffIdx` will not be consistent. That was what I was concerning.

But now I've realized, that local time sig modifications are disabled if there are linked staves. So the above scenario will never happen when we have parts.

Thanks for your correction and I'm closing this... You can submit your own PR for this bug :)

**Pull request closed without merging**